jsprit-core/src/main/java/com/graphhopper/jsprit/core/problem/job/Job.java

```diff
@@ -54,12 +61,39 @@
     public String getName();
 
     /**
-     * Get priority of job. Only 1 = high priority, 2 = medium and 3 = low are allowed.
+     * Get priority of job. Only 1 = high priority, 2 = medium and 3 = low are
+     * allowed.
      * <p>
      * Default is 2 = medium.
      *
      * @return priority
      */
     public int getPriority();
 
+
+    /**
+     * @return All involved locations
+     */
+    public List<Location> getAllLocations();
+
+    /**
+     * @return Start location
+     */
+    public Location getStartLocation();
+
+    /**
+     * @return End location
+     */
+    public Location getEndLocation();
+
+    /**
+     * @return All activities
+     */
+    public JobActivityList getActivityList();
```

Comment:
I'm not really sure what did you mean here, but yes. I think ActivityList is the main implementation for accessing all or any (fitered) set of activities of a Job.
Activity has a reference to its job, and Job has reference to all its activities. 
I guess JobActivityList will be extended with several functions before the end.
Also this abstraction allows to hide the mode we store and handle the activities of a job. (As we have two implementations already.)